operator/src/main/java/oracle/kubernetes/operator/WaitForReadyStep.java

```diff
@@ -239,6 +266,25 @@ public NextAction apply(Packet packet) {
 
   }
 
+  private class MakeRightDomainStep extends DefaultResponseStep {
+    private final Callback callback;
+
+    MakeRightDomainStep(Callback callback, Step next) {
+      super(next);
+      this.callback = callback;
+    }
+
+    @Override
+    public NextAction onSuccess(Packet packet, CallResponse callResponse) {
+      MakeRightDomainOperation makeRightDomainOperation =
+              (MakeRightDomainOperation)packet.get(MAKE_RIGHT_DOMAIN_OPERATION);
+      makeRightDomainOperation.setLiveInfo(new DomainPresenceInfo((Domain)callResponse.getResult()));
+      callback.fiber.terminate(null, packet);
```

Comment:
we probably need to call removeCallback here.

Reply:
I made the change to remove callback here.